bin/reth/src/node/cl_events.rs

```diff
@@ -0,0 +1,64 @@
+use futures::Stream;
+use reth_provider::CanonChainTracker;
+use std::{
+    pin::Pin,
+    task::{ready, Context, Poll},
+    time::Duration,
+};
+use tokio::time::Interval;
+
+/// Period of not exchanging transition configurations with consensus client,
+/// after which the warning is issued.
+const NO_TRANSITION_CONFIG_EXCHANGED_PERIOD: Duration = Duration::from_secs(120);
+/// Period of not receiving fork choice updates from consensus client,
+/// after which the warning is issued.
+const NO_FORKCHOICE_UPDATE_RECEIVED_PERIOD: Duration = Duration::from_secs(120);
+
+pub struct ConsensusLayerHealthEvents<CC> {
+    interval: Interval,
+    canon_chain: CC,
```

Comment:
I think we can actually box this and get rid of all the generics, should be fine and would make it easier

Reply:
hmm not sure, imo generics don't bring a lot of mess in this case, but having a `Box<dyn CanonChainTracker>` will be a smol perf hit

Reply:
> but having a Box<dyn CanonChainTracker> will be a smol perf hit

we're only calling this every interval, so I don't think this really matters.

I'm fine with both

Reply:
fair, boxed